### PR TITLE
Fix benchmarks and add multiple mini benchmark tests

### PR DESCRIPTION
**Pull request by liufengyun**
The benchmarks has been down due to some weird error. A single line change fix it: [link](https://github.com/lampepfl/dotty/compare/master...dotty-staging:bench?expand=1#diff-28f8d722cae3da74d903fc63040c9d36R68)

```
-  @Test def pickle_pickleOK = compileDir(testsDir, "pickling", testPickling)
+  @Test def pickle_pickleOK = compileFiles(testsDir + "pickling/", testPickling)
```

And I've added several mini tests which covers implicit search, exhaustivity check, tailrec, etc. We can just put any mini test to the folder `bench/tests`, it will be automatically tracked in the benchmarks.

### CODE REVIEW

**Comment by liufengyun** [comment]
Review @felixmulder and @DarkDimius ?

File: bench/src/test/scala/Benchmarks.scala [hunk 1]
```diff
@@ -74,11 +76,24 @@ object BenchTests extends OnlineRegressionReport {
   def setup =
     performance of "dotty" in {
       measure.method("stdlib") in {
-        using(Gen.unit("test")) curve "stdlib" in { r => stdLib }
+        // maybe scalac curve later
+        using(Gen.unit("test")) curve "dotty" in { r => stdLib }
```
Comment (liufengyun) [inline bench/src/test/scala/Benchmarks.scala]: @DarkDimius The `curve` is to support multiple curves in a chart. Yes, I should keep it the same for `stdlib` and `dotty-src` for backward data compatibility.

Comment (liufengyun) [inline bench/src/test/scala/Benchmarks.scala]: @DarkDimius the name for `stdlib` and `dotty-src` are now restored.

**Comment by odersky** [comment]
Status?

**Comment by liufengyun** [comment]
@DarkDimius  This is one is up to you to review, thanks :)

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**